CONTRIBUTING.md

```diff
@@ -0,0 +1,269 @@
+
+Please follow these steps to have your contribution considered by the maintainers:
+
+1. Follow all instructions in [the template](.github/pull_request_template.md)
+2. Follow the [styleguides](#styleguides) where applicable
+3. After you submit your pull request, verify that all [status checks](https://help.github.com/articles/about-status-checks/) are passing <details><summary>What if the status checks are failing?</summary>If a status check is failing, and you believe that the failure is unrelated to your change, please leave a comment on the pull request explaining why you believe the failure is unrelated. A maintainer will re-run the status check for you. If we conclude that the failure was a false positive, then we will open an issue to track that problem with our status check suite.</details>
+
+While the prerequisites above must be satisfied prior to having your pull request reviewed, the reviewer(s) may ask you to complete additional design work, tests, or other changes before your pull request can be ultimately accepted.
+
+### Documentation Contributions
+
+ownCloud is very proud of the documentation it has, which is the work of a great team of people. Of course, also the documentation is open to contributions.
+
+See the [Getting Started Guide](https://github.com/owncloud/docs/blob/master/docs/getting-started.md) on how to get started. Many other useful information is summarized in the [Documentation Readme](https://github.com/owncloud/docs).
+
+### Internationalization
+
+ownCloud is translated in many languages to allow people from all oever the world to use ownCloud in their native language. For translations, ownCloud uses [Transifex](https://www.transifex.com) as community based collaboration collaboration platform for internationalization.
+
+For contributions please refer to the [Transifex Resources](https://www.transifex.com/resources/) to learn how to improve ownClouds translations there.
+
+### Deployments: Docker, Ansible and friends
+
+Depending on the ownCloud component, there is complex deployment tooling to install in various environments. There is for example [ownCloud Ansible](https://github.com/owncloud-ansible) with Ansible resources. Contributions to that are very appreciated and follow the same guidelines as every other code contribution.
+
+## Styleguides
+
+To keep up with a consistent code and tooling landscape, some of the ownCloud modules maintain styleguides for contributions. It is mandatory to follow them in contributions.
+
+### Git Commit Messages
+
+* Use the present tense ("Add feature" not "Added feature")
+* Use the imperative mood ("Move cursor to..." not "Moves cursor to...")
+* Limit the first line to 72 characters or less
+* Reference issues and pull requests liberally after the first line
+* When only changing documentation, include `[docs-only]` in the commit title
+
+### Golang Styleguide
+
+TBD
```

Comment:
golang comes with a built in formatter. that part at least is not discussed. There is also https://golang.org/doc/effective_go and this talk by rob pike about why simplicity may be the reason for success of golang: https://www.youtube.com/watch?v=rFejpH_tAHM